### DISCUSSION TITLE
Logic of the text_encoder_training with Resume_Training

### DISCUSSION THREAD

**Discussion by Pasha49** [TheLastBen/fast-stable-diffusion#575] [Q&A] [answered]
Hello! Please tell me how it will work?

If I want to train 10,000 steps, but with travel_text_encoder_for = 30%, and time limit for free training. How to do it?
In one training session, I can take 2500 steps.
30% of 10,000 = 3,000.

1)
- 1st training(First_Training): 2,500 st and travel_text_encoder_for=100   (2,500 text_encoder) 
- 2nd training(Resume_Training): 2500 st and travel_text_encoder_for=20   (500 text_encoder) 
- 3rd training(Resume_Training): 2500 st and travel_text_encoder_for=0   (0 text_encoder)
- 4th training(Resume_Training): 2500 st and travel_text_encoder_for=0   (0 text_encoder) 

or

2)
- 1st training(First_Training): 2,500 st and travel_text_encoder_for=30   (750 text_encoder) 
- 2nd training(Resume_Training): 2500 st and travel_text_encoder_for=30   (750 text_encoder)
- 3rd training(Resume_Training): 2500 st and travel_text_encoder_for=30   (750 text_encoder)
- 4th training(Resume_Training): 2500 st and travel_text_encoder_for=30   (750 text_encoder)

**Comment by TheLastBen** [comment, marked as answer]
Spreading the textenc finetuning through out the training steps should yield better results, go with the second one

**Reply by Pasha49** [reply]
Thank you! For two characters + the ability to use styles, how much percent will you advise?

**Reply by TheLastBen** [reply]
the characters must be from different genders, the text encoder must not go over 1000 steps, no matter the total training steps, then continue training the model without the text_enc until you get the desired result.

**Reply by Pasha49** [reply]
Thank you!

Total, for 2 characters of different genders. 15 photos on the character. Total photos 30.
30*200 = 6000 steps.
Maximum 1000 steps with Text Encoder.


-  1st training(First_Training): 2,500 st and travel_text_encoder_for=17 (425 text_encoder)
-  2nd training(Resume_Training): 2500 st and travel_text_encoder_for=17 (425 text_encoder)
-  3rd training(Resume_Training): 1000 st and travel_text_encoder_for=17 (170 text_encoder)
Total 6000 steps, 1000 text encoder.

that's right?

**Reply by TheLastBen** [reply]
sounds right

**Reply by TheLastBen** [reply]
if you don't get the desired results, resume training for 200 steps text_encoder
